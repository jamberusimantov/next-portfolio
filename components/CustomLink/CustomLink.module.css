.disabled {
    pointer-events: none;
}

.link {
    color: var(--blue);
}