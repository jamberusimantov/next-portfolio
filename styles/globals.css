html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
    opacity: 1;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.submitBtn {
    width: 100%;
    max-width: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: .5rem;
}

.submitBtn p {
    flex: 1;
    margin: 0;
    font-size: large;
    padding: 0 !important;
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

@media(min-width:580px) {
    .submitBtn {
        margin: auto;
    }
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}