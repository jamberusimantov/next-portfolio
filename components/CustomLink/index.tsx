import React, { forwardRef, ReactElement } from "react";
import styles from "./CustomLink.module.css";
import Link from 'next/link';


interface iCustomLinkProps {
    children?: ReactElement | ReactElement[] | string,
    href?: string,
    target?: keyof { "_blank": "" },
    title?: string,
    disabled?: boolean,
    testid?: string,
    className?: string,
    prefetch?: boolean 
};

const CustomLink = forwardRef<HTMLAnchorElement, iCustomLinkProps>((props, ref) =>
    <Link
        href={props.href ?? "/path/to/somewhere"}
        shallow
        replace
        passHref
        prefetch={props.prefetch ?? false}
        children={<a
            data-testid={props.testid}
            ref={ref}
            target={props.target}
            title={props.title}
            children={props.children?? "/path/to/somewhere"}
            className={props.disabled ? styles.disabled : props.className ?? styles.link}
        />
        }
    />
);

export default CustomLink;