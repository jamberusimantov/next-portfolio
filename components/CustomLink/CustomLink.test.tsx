/**
 * @jest-environment jsdom
 */
import React from "react";
import { describe, expect, test } from '@jest/globals';
import { render, act } from '@testing-library/react';
import CustomLink from '.';


describe('CustomLink', () => {
    test('use jsdom in this test file', () => {
        expect(document.createElement('div')).not.toBeNull();
    });
    test('renders correctly', async () => {
        let el: HTMLElement = document.createElement("div");
        let getText: Function = () => { };
        let getTestId: Function = () => { };
        await act(async () => {
            let { container, getByText, getByTestId } = await render(<CustomLink testid="CustomLink"/>);
            el = container;
            getText = getByText;
            getTestId = getByTestId;
        });
        expect(el).toMatchSnapshot();
        expect(getText("/path/to/somewhere"));
        getTestId("CustomLink").textContent = "/somewhere/to/path";
        expect(getText("/somewhere/to/path"));
    });
});