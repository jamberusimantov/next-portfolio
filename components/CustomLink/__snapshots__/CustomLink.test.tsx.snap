// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLink renders correctly 1`] = `
<div>
  <a
    class="link"
    data-testid="CustomLink"
    href="/path/to/somewhere"
  >
    /path/to/somewhere
  </a>
</div>
`;
